### ISSUE TITLE
MaxNLocator.default_params deprecation may break Cartopy

### ISSUE THREAD

**Issue by QuLogic** [matplotlib/matplotlib#13991]
At least, according to @jklymak in https://github.com/SciTools/cartopy/pull/1299. I don't have a reproducer, but I'm opening this so we can avoid another problem like `ax._hold` in 3.0.0.

**Comment by jklymak** [comment]
Ping @timhoffm.  I’m thoroughly confused by this and I’m not sure if Cartopy should ever have done this, but ...

**Comment by jklymak** [comment]
Basically cartopy does

```python
import matplotlib.ticker as mticker

class FancyLocator(mticker.MaxNLocator):
    default_params = mticker.MaxNLocator.default_params.copy()
    default_params.update(nbins=8, dms=False)

loc0 = mticker.MaxNLocator()
loc1 = FancyLocator()
```

running this gives:

```
Traceback (most recent call last):
  File "testLocator.py", line 3, in <module>
    class FancyLocator(mticker.MaxNLocator):
  File "testLocator.py", line 4, in FancyLocator
    default_params = mticker.MaxNLocator.default_params.copy()
AttributeError: '_deprecated_property' object has no attribute 'copy'
```

**Comment by dstansby** [comment]
I looks to me like adding the deprecation decorator means that the property can't be copied?

**Comment by jklymak** [comment]
I don’t think properties can be copied at all, can they?

**Comment by tacaswell** [comment]
Sorry if this is redundent with other analysis of this

Looks like this came in via f084e0d262eae77e63d123c1b29594df751664e6

We used to have a class-level dict `default_params`, we made that private and put in a property to warn on user access, however, that only works at the instance scope. 

cartpy was defining a class that sub classed MaxNLocator, copied the defaults and then updated the copy.  On its face that seems ok.
